src/system/calculate_quadrature_point_values.hpp

```diff
@@ -58,14 +61,15 @@ struct CalculateQuadraturePointValues {
         const auto i_elem = static_cast<size_t>(member.league_rank());
         const auto num_qps = num_qps_per_element(i_elem);
 
-        Kokkos::parallel_for(
-            Kokkos::TeamThreadRange(member, num_qps), CalculateRR0{i_elem, qp_r0_, qp_r_, qp_RR0_}
-        );
         Kokkos::parallel_for(
             Kokkos::TeamThreadRange(member, num_qps),
             CalculateTemporaryVariables{i_elem, qp_x0_prime_, qp_u_prime_, qp_x0pupSS_}
         );
         member.team_barrier();
+        Kokkos::parallel_for(
+            Kokkos::TeamThreadRange(member, num_qps), CalculateRR0{i_elem, qp_x_, qp_RR0_}
+        );
+        member.team_barrier();
```

Comment:
Why did you change the order of `CalculateTemporaryVariables` and `CalculateRR0` and add the new team_barrier?  I'm not seeing the change in dependencies.

Reply:
That's a good catch. I had made the change in the first commit and the second commit made the change unnecessary. I had added the function to calculate QP position, `qp_x` right before this, which made `CalculateRR0` depend on that function's completion. So I added the team barrier to guarantee that the positions were all calculated before moving on. Then I ended up moving the calculate QP position call which made this change unnecessary. I'll fix it in the next commit.